### PR TITLE
Add listSubscriptions to resourceApi typings

### PR DESCRIPTION
**Pull request by nturinski**
Implementation needs to be done in Resource Groups, which will be coming soon

### CODE REVIEW

File: utils/hostapi.d.ts [hunk 1]
```diff
@@ -61,6 +61,13 @@ export interface AzureHostExtensionApi {
      */
     registerActivity(activity: Activity): Promise<void>;
 
+    /**
+     *
+     * @param filter Whether to filter the list returned, according to the list returned
+     * by `getTenantFilters()` and `getSubscriptionFilters()`. Optional, default true.
+     */
+    getSubscriptions: (filter: boolean) => Promise<AzureSubscription[]>;
```
Comment (MicroFish91) [inline utils/hostapi.d.ts:69]: Should this be `(filter?: boolean)`?

Comment (nturinski) [reply utils/hostapi.d.ts:69]: Oh yeah, I guess the comment does say that. I kind of wanted to be explicit about it so I'll amend the comment.

**Review by MicroFish91 — APPROVED** [review]

**Comment by bwateratmsft** [comment]
Shouldn't the auth package be used instead?

**Comment by nturinski** [comment]
Maybe... Here's my scenario:

I need a way to list the filtered subscriptions to create my own `SubscriptionListStep` for a user to select their sub. This is happening in the Functions and Storage extension.  Since I don't think we'd want the client extensions to directly consume the auth package, I was planning on having Resources handle listing the subs out to the clients.

Resources also controls the tenant/subscription filter settings, so I thought it made more sense for it to deliver the list to the clients.

**Pull request merged**